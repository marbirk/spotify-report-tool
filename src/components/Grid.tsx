import React from 'react'

interface GridProps {
    children: React.ReactNode
}

const Grid = ({ children, ...props }: GridProps) => (
    <div
        className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-4"
        {...props}
    >
        {children}
    </div>
)

export default Grid
