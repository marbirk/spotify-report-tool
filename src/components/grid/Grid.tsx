import React from 'react'

interface GridProps {
    children: React.ReactNode
}

const Grid = ({ children, ...props }: GridProps) => (
    <div
        className="grid grid-cols-2 sm:grid-cols-2 md:grid-cols-3 gap-4 mt-6"
        {...props}
    >
        {children}
    </div>
)

export default Grid
