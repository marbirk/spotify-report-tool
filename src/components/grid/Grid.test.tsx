import React from 'react'
import { render, screen } from '@testing-library/react'

import Grid from './Grid'

describe('Grid', () => {
    test('renders correctly', () => {
        render(<Grid>Grid children</Grid>)
        const component = screen.getByText('Grid children')
        expect(component).toBeInTheDocument()
        expect(component).toMatchSnapshot()
    })
})
