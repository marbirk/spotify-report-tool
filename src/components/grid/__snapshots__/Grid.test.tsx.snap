// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
<div
  class="grid grid-cols-2 sm:grid-cols-2 md:grid-cols-3 gap-4 mt-6"
>
  Grid children
</div>
`;
